applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -288,32 +271,45 @@ void ElasticIsotropic3D::CalculateElasticMatrix(
 void ElasticIsotropic3D::CalculatePK2Stress(
     const Vector& rStrainVector,
     Vector& rStressVector,
-    const double E,
-    const double NU
+    ConstitutiveLaw::Parameters& rValues
 )
 {
-    SizeType SizeSystem = GetStrainSize();
-    Matrix C(SizeSystem,SizeSystem);
-    CalculateElasticMatrix(C, E, NU);
-    noalias(rStressVector) = prod(C,rStrainVector);
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
```

Comment:
Why do you reimplement this, the other version seems shorter to me

Reply:
regarding the comment of @RiccardoRossi about the efficiency of the computation - to compute without allocate the matrix C